### PR TITLE
Report page of problems uploading documents

### PR DESCRIPTION
**Pull request by zheileman**
This is a simple reporting page, listing the problems users had uploading
documents when they select said checkbox. This is what we call a postal
route as users can't continue to submit the case to GLiMR and we instruct
them to send the documentation by post instead.

The page is protected by `http basic auth`, and the digest of the password
is set as an ENV variable called `UPLOAD_PROBLEMS_REPORT_AUTH_DIGEST`

https://www.pivotaltracker.com/story/show/147615635

### CODE REVIEW

**Comment by digitalronin** [comment]
We're only authenticating based on the password, here.

I don't think it makes a lot of difference, in terms of security, but I think people might be more accustomed to having pairs of credentials - i.e. both a username and a password.

Any thoughts?

File: app/controllers/admin/upload_problems_report_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+  def index
+    @cases_with_upload_problems = TribunalCase.with_upload_problems.order(created_at: :desc)
+  end
+
+  private
+
+  def authenticate
+    authenticate_or_request_with_http_basic do |_user, password|
+      Digest::SHA1.hexdigest(password) == ENV.fetch('UPLOAD_PROBLEMS_REPORT_AUTH_DIGEST')
```
Comment (digitalronin) [inline app/controllers/admin/upload_problems_report_controller.rb]: Is there any reason not to use sha256 here? I know it's overkill, but as a general rule I don't think we should be depending on sha1, these days.

**Comment by zheileman** [comment]
@digitalronin yep, http auth will ask for both regardless, but I decided to ignore on purpose the user bit as it doesn't really matter to us in this specific context and page. Also, in terms of security, the user otherwise would have to be also stored in the ENV, unless we decide to just hardcode a user, 'admin' for instance in the code. Do you think we should enforce a specific user/password? I don't mind changing it.

File: app/controllers/admin/upload_problems_report_controller.rb (hunk 1 above)
Comment (zheileman) [inline app/controllers/admin/upload_problems_report_controller.rb]: I will change it to sha256.

**Comment by zheileman** [comment]
I made changes to use both user/password pair, and SHA256.

**Pull request merged**